src/fetchers/tarball-fetcher.js

```diff
@@ -273,6 +276,24 @@ export default class TarballFetcher extends BaseFetcher {
     }
   }
 
+  requestHeaders(requestUrl: string): {[string]: string} {
```

Comment:
I based this method off of https://github.com/yarnpkg/yarn/blob/b8a565e562810f9bf86f3ac0531ad348ed7e9d0a/src/registries/npm-registry.js#L200-L213

Although, looking at it now I think this `requestHeaders` does not need a `requestUrl` argument and instead could use `this.reference` directly.